python/pyarrow/_compute.pyx

```diff
@@ -690,6 +690,26 @@ cdef class _CastOptions(FunctionOptions):
 
 
 class CastOptions(_CastOptions):
+    """
+    Options for the `cast` function.
+
+    Parameters
+    ----------
+    target_type : DataType, optional
+        The PyArrow type to cast to.
+    allow_int_overflow : bool, optional
+        Whether integer overflow is allowed when casting.
```

Comment:
All those options indicate "optional", but I suppose they have an actual default? (all False?)

Reply:
Indeed, all False by default (in C++).